onadata/apps/logger/models/version.py

```diff
@@ -0,0 +1,51 @@
+from django.db import models
+
+
+class VersionTreeException(Exception):
+    pass
+
+
+class VersionManager(models.Manager):
```

Comment:
I'd rewrite it as follows:
- get_root_path should return reversed path [path[0] == root, path[1].parent == root, ..., path[-1] == vt]
- with above condition met, find_path throws exception if vt1_root_path[0] != vt2_root_path[0]; (or any of them is empty - but then you'd have to check if that state is even possible)
- otherwise - there is a path, and you just have to find first index in range(len(vt2_root_path) - 1) where vt1_root_path[index] == vt2_root_path[index] and vt1_root_path[index + 1] != vt2_root_path[index + 1]. Note - you could even binsearch for such point, but imo right now it's not worth the effort.

Benefits of the above approach:
- you don't have to do in operation on a list (slow - your current approach is O(n^2) because of that)
- seems a bit more clear when exception is thrown

And I'd double check if VersionTreeException is even needed - assuming you properly save things to DB it should never happen (of course I'm aware expectation != reality, but still...)